projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/datajobs/it/PrivateBuilderDockerRepoIT.java

```diff
@@ -0,0 +1,218 @@
+    String dataJobRequestBody = getDataJobRequestBody(TEST_TEAM_NAME, TEST_JOB_NAME);
+
+    // Execute create job
+    mockMvc
+        .perform(
+            post(String.format("/data-jobs/for-team/%s/jobs", TEST_TEAM_NAME))
+                .with(user("user"))
+                .content(dataJobRequestBody)
+                .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(status().isCreated());
+  }
+
+  @Test
+  public void testPrivateDockerBuildJob() throws Exception {
+    createBuilderImagePullSecret(getControlNamespace());
+    // Take the job zip as byte array
+    byte[] jobZipBinary =
+        IOUtils.toByteArray(
+            getClass().getClassLoader().getResourceAsStream("job_ephemeral_storage.zip"));
+
+    // Execute job upload with user
+    MvcResult jobUploadResult =
+        mockMvc
+            .perform(
+                post(String.format(
+                        "/data-jobs/for-team/%s/jobs/%s/sources", TEST_TEAM_NAME, TEST_JOB_NAME))
+                    .with(user("user"))
+                    .content(jobZipBinary)
+                    .contentType(MediaType.APPLICATION_OCTET_STREAM))
+            .andExpect(status().isOk())
+            .andReturn();
+
+    DataJobVersion testDataJobVersion =
+        new ObjectMapper()
+            .readValue(jobUploadResult.getResponse().getContentAsString(), DataJobVersion.class);
+    Assertions.assertNotNull(testDataJobVersion);
+
+    String testJobVersionSha = testDataJobVersion.getVersionSha();
+    Assertions.assertFalse(StringUtils.isBlank(testJobVersionSha));
+
+    // Setup
+    String dataJobDeploymentRequestBody = getDataJobDeploymentRequestBody(testJobVersionSha);
+
+    // Execute build and deploy job
+    mockMvc
+        .perform(
+            post(String.format(
+                    "/data-jobs/for-team/%s/jobs/%s/deployments", TEST_TEAM_NAME, TEST_JOB_NAME))
+                .with(user("user"))
+                .content(dataJobDeploymentRequestBody)
+                .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(status().isAccepted())
+        .andReturn();
+
+    String opId = TEST_JOB_NAME + UUID.randomUUID().toString().toLowerCase();
+
+    // manually start job execution
+    mockMvc
+        .perform(
+            post(String.format(
+                    "/data-jobs/for-team/%s/jobs/%s/deployments/%s/executions",
+                    TEST_TEAM_NAME, TEST_JOB_NAME, TEST_JOB_DEPLOYMENT_ID))
+                .with(user("user"))
+                .header(HEADER_X_OP_ID, opId)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(
+                    "{\n"
+                        + "  \"args\": {\n"
+                        + "    \"key\": \"value\"\n"
+                        + "  },\n"
+                        + "  \"started_by\": \"schedule/runtime\"\n"
+                        + "}"))
+        .andExpect(status().is(202))
+        .andReturn();
+
+    // wait for pod to initialize
+    Thread.sleep(4000);
```

Comment:
I don't get that

Reply:
We just don't check when we know the pod is defo not up. 
It keeps the logs a lot cleaner.